plugins/innogrit/innogrit-nvme.c

```diff
@@ -0,0 +1,534 @@
+	};
+
+	OPT_ARGS(opts) = {
+		OPT_UINT("namespace-id", 'n', &cfg.namespace_id,   namespace),
+		OPT_END()
+	};
+
+	fd = parse_and_open(argc, argv, desc, opts);
+	if (fd < 0)
+		return fd;
+
+	nvme_get_log_smart(fd, cfg.namespace_id, true, &smart_log);
+	nvme_show_smart_log(&smart_log, cfg.namespace_id, devicename, NORMAL);
+
+	printf("DW0[0-1]  Defect Cnt                    : %u\n", pvsc_smart->defect_cnt ); 
+	printf("DW0[2-3]  Slc Spb Cnt                   : %u\n", pvsc_smart->slc_spb_cnt); 
+	printf("DW1       Slc Total Ec Cnt              : %u\n", pvsc_smart->slc_total_ec_cnt);
+	printf("DW2       Slc Max Ec Cnt                : %u\n", pvsc_smart->slc_max_ec_cnt);
+	printf("DW3       Slc Min Ec Cnt                : %u\n", pvsc_smart->slc_min_ec_cnt);
+	printf("DW4       Slc Avg Ec Cnt                : %u\n", pvsc_smart->slc_avg_ec_cnt);
+	printf("DW5       Total Ec Cnt                  : %u\n", pvsc_smart->total_ec_cnt);
+	printf("DW6       Max Ec Cnt                    : %u\n", pvsc_smart->max_ec_cnt);
+	printf("DW7       Min Ec Cnt                    : %u\n", pvsc_smart->min_ec_cnt);
+	printf("DW8       Avg Ec Cnt                    : %u\n", pvsc_smart->avg_ec_cnt);
+	printf("DW9       Mrd Rr Good Cnt               : %u\n", pvsc_smart->mrd_rr_good_cnt);
+	printf("DW10      Ard Rr Good Cnt               : %u\n", pvsc_smart->ard_rr_good_cnt);
+	printf("DW11      Preset Cnt                    : %u\n", pvsc_smart->preset_cnt);
+	printf("DW12      Nvme Reset Cnt                : %u\n", pvsc_smart->nvme_reset_cnt);
+	printf("DW13      Low Pwr Cnt                   : %u\n", pvsc_smart->low_pwr_cnt);
+	printf("DW14      Wa                            : %u\n", pvsc_smart->wa);
+	printf("DW15      Ps3 Entry Cnt                 : %u\n", pvsc_smart->ps3_entry_cnt);
+	printf("DW16[0]   highest_temp[0]               : %u\n", pvsc_smart->highest_temp[0]); 
+	printf("DW16[1]   highest_temp[1]               : %u\n", pvsc_smart->highest_temp[1]); 
+	printf("DW16[2]   highest_temp[2]               : %u\n", pvsc_smart->highest_temp[2]); 
+	printf("DW16[3]   highest_temp[3]               : %u\n", pvsc_smart->highest_temp[3]); 
+	printf("DW17      weight_ec                     : %u\n", pvsc_smart->weight_ec);
+	printf("DW18      slc_cap_mb                    : %u\n", pvsc_smart->slc_cap_mb);
+	printf("DW19-20   nand_page_write_cnt           : %llu\n", pvsc_smart->nand_page_write_cnt);
+
+	iIndex = 21;
+	for(i = 0; i < (sizeof(pvsc_smart->reserved2) /4); i++)
+	{
+		if(pvsc_smart->reserved2[i] != 0)
+			printf("DW%-2d                                    : %u\n", iIndex, pvsc_smart->reserved2[i]);
+		iIndex++;
+	}
+
+	return 0;
+}
+
+
+
+
+/*****************************************************************************
+*
+* ENTRY POINT: SortEventLogFn
+*
+*****************************************************************************/
+int SortEventLogFn(const void * a, const void * b)
+{
+	int rc = 0;
+	if (((PEVENTLOG_ADDINDEX)a)->ms > ((PEVENTLOG_ADDINDEX)b)->ms)
```

Comment:
Instead casting 6 times in this function do it once at the beginning of the fuction (these type names look horrible. WRITING ALL IN CAPITAL LETTERS IS DISCOURAGED). 

Here my version of this code function

```c
struct eventlog_addindex {
	unsigned int ms;
	unsigned int param[7];
	unsigned int index;
};

static int sort_event_log_cb(const void * a, const void * b)
{
	const struct eventlog_addindex *eva = a, *evb = b;

	if (eva->ms == evb->ms) {
		if (eva->index < evb->index)
			return -1;
		else
			return 0;
	}

	if (eva->ms > evb->ms)
		return 1;
	/* eva->ms < evb->ms */
	return -1;
}
```

No need for typedefs either. Please note we do not prefix the variables with the types (`iIndex` vs `index`)